web/src/pages/api/user/confirmEmail.ts

```diff
@@ -0,0 +1,59 @@
+import { User } from "@prisma/client";
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { prisma } from "../../../server/db/client";
+import { validateJWT } from "../../../utils/jwt";
+import { validateMethod } from "../../../utils/validation";
+
+export default async function confirmEmail(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (!validateMethod("GET", req, res)) return;
+
+  const userID = req.query.id;
+  const token = req.query.token;
+
+  if (
+    Array.isArray(userID) ||
+    userID === undefined ||
+    Array.isArray(token) ||
+    token === undefined
+  ) {
+    return res
+      .status(400)
+      .json({ Error: "Passed params cannot be a string array or undefined" });
+  }
+
+  const isJWTValid: boolean = await validateJWT(userID, token);
+  if (!isJWTValid) {
+    return res.status(400).json({ Error: "Token is invalid." });
+  }
+
+  // verify that the user exists with the userID
+  const maybeUser: User | null = await prisma.user.findFirst({
+    where: {
+      id: userID,
+    },
+  });
+
+  // if user does exist, then create a date object and add that user's emailVerified date
+  if (maybeUser) {
+    await prisma.user.update({
+      where: {
+        id: maybeUser.id,
+      },
+      data: {
+        emailVerified: new Date(),

```

Comment:
Have you verified that this actually works? I don't know if prisma can handle direct dates like that

Reply:
This is what I saw in the Prisma docs so I assumed it was correct but I don't know another way of doing it hmm